### PR TITLE
DISCO-656: Adds in text search param for publisher dashboard

### PR DESCRIPTION
**Pull request by jlajoie**

### CODE REVIEW

File: course_discovery/apps/api/v1/tests/test_views/test_courses.py [hunk 1]
```diff
@@ -266,6 +266,56 @@ def test_list_exclude_utm(self):
             self.serialize_course([self.course], many=True, extra_context=context)
         )
 
+    def test_list_pubq_by_title(self):
+        """ Verify the endpoint returns a list of courses filtered by title when specified with pubq and editable """
+        url = reverse('api:v1:course-list') + '?editable=1&pubq=Test'
+
+        draft = CourseFactory(partner=self.partner,
```
Comment (Dillon-Dumesnil) [inline course_discovery/apps/api/v1/tests/test_views/test_courses.py]: Nit: Switch to 
```
CourseFactory(
    partner=self.partner,
    uuid=self.course.uuid,
    key=self.course.key,
    draft=True,
    title='SomeTestTitle',
)

Comment (Dillon-Dumesnil) [reply course_discovery/apps/api/v1/tests/test_views/test_courses.py]: Also not sure if it's used in here or not, but might be able to just use `ensure_draft_world`

File: course_discovery/apps/api/v1/tests/test_views/test_courses.py [hunk 2]
```diff
@@ -266,6 +266,56 @@ def test_list_exclude_utm(self):
...
+        self.course.draft_version = draft
+        self.course.save()
+        # Create a random test course with a title without the phrase "Test" in it
+        CourseFactory(partner=self.partner, key=self.course.key + 'Z', title='RandomString')
+
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+        unique_keys = set()
+        for course_result in response.data['results']:
+            print(course_result)
```
Comment (Dillon-Dumesnil) [inline course_discovery/apps/api/v1/tests/test_views/test_courses.py]: GET OUT

**Review by Dillon-Dumesnil — APPROVED** [review]
Remove the print. Otherwise, not too bad.

File: course_discovery/apps/api/v1/tests/test_views/test_courses.py (hunk 2 above)
Comment (jlajoie) [reply course_discovery/apps/api/v1/tests/test_views/test_courses.py]: Oh no... oh no...

File: course_discovery/apps/api/v1/tests/test_views/test_courses.py (hunk 1 above)
Comment (jlajoie) [reply course_discovery/apps/api/v1/tests/test_views/test_courses.py]: Yup that worked, refactored 👍

**Pull request merged**